File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(string) {
    this.string = string;
    this._rows = string.split('\n').map(e=>e.split(' '));
    this._columns = [];
    
  }

  get rows() {
    return this._rows.map(e=>{
      return e.map(f=>{
        return Number(f);
      })
    })
  }

  get columns() {
    for (let i=0; i<this.rows[0].length;i++){
      this._columns.push(this.rows.map(e=>e[i]));
    }
    return this._columns;
  }
}

File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.rows = [ [int(x) for x in row.split(' ')] for row in matrix_string.split('\n') ]

    def row(self, index):
        return self.rows[index-1]

    def column(self, index):
        return [ self.row(x + 1)[index - 1] for x in range(len(self.rows)) ]